Updated player deletion code to accommodate tables that do not use the column name citizenid

tlsharkie opened the pull request:
Updated player deletion code in `server/player.lua` to accommodate tables that do not use the column name `citizenid`.
Many scripts use other column names to store the `citizenid` but contain data that should be deleted upon character deletion. This leads to excess data build up in the database that is no longer needed and currently requires manual effort to clean out the database.

New format for `playertables` now contains table and key variables (primary key column name)
and updated SQL commands in `DeleteCharacter` and `ForceDeleteCharacter` to handle variable column names

```
--- 'table' is the new SQL table name
--- 'key' is the column name which contains the 'citizenid'
{ table = 'players', key = 'citizenid' } 
```

## Checklist

<!-- Put an x inside the [ ] to check an item, like so: [x] -->

- [x] I have personally loaded this code into an updated qbcore project and checked all of its functionality.
- [x] My code fits the style guidelines.
- [x] My PR fits the contribution guidelines.

tlsharkie committed:
Update server/player.lua

Updated `Delete character` code to handle database tables where primary key is not `citizenid`

Changed `playertables` format to handle any primary key

Updated `DeleteCharater()` and `ForceDeleteCharater()` SQL commands

tlsharkie reviewed (commented):
removed trailing whitespaces

tlsharkie reviewed (commented):
removed trailing whitespaces from functions.lua to pass lint tests

tlsharkie commented:
updated code and removed trailing whitespaces to pass lint tests

Z3rio reviewed (changes requested):
Overall looks good to me, just some tables that shouldnt be there, otherwise I'd be fine to approve this :)

tlsharkie commented on server/player.lua:
```diff
@@ -529,42 +529,65 @@ end
+
+     --Tables where primary key = 'identifier'
+    { table = 'player_contacts', key = 'identifier' },
+    { table = 'bank_cards', key = 'identifier' },
+    { table = 'bank_history', key = 'identifier' },
+    { table = 'casino_players', key = 'identifier' },
+    { table = 'cd_garage_privategarage', key = 'identifier' },
+    { table = 'inventory_clothes', key = 'identifier' },
+    { table = 'lation_detecting', key = 'identifier' },
+    { table = 'm_hunting', key = 'identifier' },
+    
+    --Tables with other citizenid labels
+    { table = 'bank_process',  key = 'owner' },
+    { table = 'mail_accounts', key = 'owner' },
+    { table = 'lation_chopshop', key = 'player_identifier' },
```
I am fine with you remvoving the non-qbcore tables from the official push. 
Please use the addtional tables as a reference for how to use the new format.

Z3rio replied:
> I am fine with you remvoving the non-qbcore tables from the official push. Please use the addtional tables as a reference for how to use the new format.

This is your PR, so I cant remove them for you. Sorry

tlsharkie committed:
Update player.lua

Updated player.lua to only include default QBCore tables for the character deletion.

tlsharkie commented:
Sorry for the delay on this one. Removed non-QBCore tables from the player deletion code.

Z3rio reviewed (approved):
LGTM

tlsharkie committed:
Merge branch 'main' into main
